accepted/00083-image_management.md

```diff
@@ -0,0 +1,311 @@
+   so the file name will still contain the build_id in order to make it unique.
+   For details see the Image building workflow below.
+
+
+```
+
+images:
+    POS_Image_Graphical7:  # <name>
+        7.0.0-1:           # <version>-<revision>
+            arch: x86_64
+            basename: POS_Image_Graphical7.x86_64-7.0.0
+            boot_image: POS_Image_Graphical7-7.0.0
+    ...
+            sync:
+    ...
+                bundle_url: https://server.tf.local/os-images/1//POS_Image_Graphical7.x86_64-7.0.0-build33.tar.xz
+
+```
+
+
+3. Do not re-use `suseImageInfo` entries, create new entry for each build.
+
+   Add yes_no column `obsolete` to `suseImageInfo`.
+
+   Drop table `suseImageBuildHistory`.
+
+   Change foreign key of `suseImageRepoDigest` table from `suseImageBuildHistory.id` to `suseImageInfo.id`.
+
+   For Docker the store can hold only one `name-version` image. Pushing new revision
+   to the store replaces the old one. So the old `suseImageInfo` entries can be set to `obsolete` when the
+   new image is built successfully.
+   On failed build the store still contains the previous revision and the corresponding entry
+   in `suseImageInfo` stays unchanged.
+
+   For details see the Image building workflow below.
+
+   For Kiwi it is possible to have multiple revisions of the same `name-version`. So the entries
+   can be deleted individually. Typically an old image can be deleted when the new image is
+   synced to all Branch servers.
+
+   To maintain previous functionality, add functions to 'delete all revisions'
+   and 'delete all obsolete revisions' of an image.
+
+   `suseImageBuildHistory` will need special attention during migration - it will be necessary
+   to create new row in `suseImageInfo` for each row from `suseImageBuildHistory`
+
+4. Add `pillar_id` to `suseImageInfo`.
+
+   Add table `suseImageFiles` with columns
+   - `id`
+   - `image_id`
+   - `file`
+   - `type`
+   - `external`
+
+   `pillar_id` points to pillar ID in `suseSaltPillars` table.
+
+   Note: According to the RFC https://github.com/uyuni-project/uyuni-rfc/blob/master/accepted/00080-jsonb-pillar-db-storage.md
+   there is a constraint `UNIQUE (org_id, category)` in `suseSaltPillars`.
+   The `category` for image pillars will be set to `"image_" + suseImageInfo.id`
+
+   `suseImageFiles` contains file or files related to an image.
+   At the beginning there will be only one file of type `bundle`
+   allowed. Later we can extend it to support multiple files per image
+   with types `kernel`, `initrd`, etc.
+
+   `file` is a local path for images in SUMA Store or URL for
+   external images (determined by `external` flag).
+
+   Pillars and local image files will be deleted when the image is deleted
+   in UI or API.
+
+   `file` will be used to show image URL in UI.
+
+   Pillar entries will be deleted when the corresponding `suseImageInfo` entry is deleted.
+   This will be implemented with an sql trigger.
+
```

Comment:
Same for `suseImageFiles` entries. But in that case `ON DELETE CASCADE` can be used.

Reply:
This is common practice, no need to mention it explicitly, IMHO.

Reply:
I disagree, you mentioned pillars and deltas, why not mention that `suseImageFiles` entries would be deleted as well when related `suseImageInfo` is removed. (The part with cascade sure can be omitted)